namer/namer.cc

```diff
@@ -1047,9 +1047,12 @@ class SymbolDefiner {
             return symbol;
         }
 
-        auto implicitlyPrivate = ctx.owner.enclosingClass(ctx) == core::Symbols::root();
+        // Methods defined at the top level default to private (on Object)
+        // Also, the `initialize` method defaults to private
+        auto implicitlyPrivate =
+            (ctx.owner.enclosingClass(ctx) == core::Symbols::root()) ||
+            (!symbol.data(ctx)->owner.data(ctx)->attachedClass(ctx).exists() && name == core::Names::initialize());
```

Comment:
This is the main change, and is mimicked by the Ruby VM:

https://github.com/ruby/ruby/blob/6847ee089d7655b2a0eea4fee3133aeacd4cc7cc/vm_method.c#L546-L557

(They also have support for `initialize_{copy,clone,dup}` but Sorbet doesn't have any special handling for those yet.)